zebra-rpc/src/methods.rs

```diff
@@ -190,11 +193,87 @@ where
+            .latest_chain_tip
+            .best_tip_height()
+            .ok_or_else(|| Error {
+                code: ErrorCode::ServerError(0),
+                message: "No Chain tip available yet".to_string(),
+                data: None,
+            })?;
+
+        // `best_block_hash` field
+        let tip_hash = self.latest_chain_tip.best_tip_hash().ok_or_else(|| Error {
+            code: ErrorCode::ServerError(0),
+            message: "No Chain tip available yet".to_string(),
+            data: None,
+        })?;
+
+        // `estimated_height` field
+        let estimated_height = self
+            .latest_chain_tip
+            .estimate_network_chain_tip_height(network, Utc::now())
+            .ok_or_else(|| Error {
+                code: ErrorCode::ServerError(0),
+                message: "No Chain tip available yet".to_string(),
+                data: None,
+            })?;
+
+        // `upgrades` object
+        //
+        // Get the network upgrades in height order, like `zcashd`.
+        let mut upgrades = IndexMap::new();
+        for (activation_height, network_upgrade) in NetworkUpgrade::activation_list(network) {
+            // Zebra defines network upgrades based on incompatible consensus rule changes,
+            // but zcashd defines them based on ZIPs.
+            //
+            // All the network upgrades with a consensus branch ID are the same in Zebra and zcashd.
+            if let Some(branch_id) = network_upgrade.branch_id() {
+                // zcashd's RPC seems to ignore Disabled network upgrages, so Zebra does too.
+                let status = if tip_height >= activation_height {
+                    NetworkUpgradeStatus::Active
+                } else {
+                    NetworkUpgradeStatus::Pending
+                };
+
+                let upgrade = NetworkUpgradeInfo {
+                    name: network_upgrade,
+                    activation_height,
+                    status,
+                };
+                upgrades.insert(ConsensusBranchIdHex(branch_id), upgrade);
+            }
+        }
+
+        // `consensus` object
+        let next_block_height =
+            (tip_height + 1).expect("valid chain tips are a lot less than Height::MAX");
+        let consensus = TipConsensusBranch {
+            chain_tip: ConsensusBranchIdHex(
+                NetworkUpgrade::current(network, tip_height)
+                    .branch_id()
+                    .unwrap_or(ConsensusBranchId::RPC_MISSING_ID),
+            ),
+            next_block: ConsensusBranchIdHex(
+                NetworkUpgrade::current(network, next_block_height)
+                    .branch_id()
+                    .unwrap_or(ConsensusBranchId::RPC_MISSING_ID),
+            ),
+        };
+
         let response = GetBlockChainInfo {
-            chain: "TODO: main".to_string(),
+            chain,
+            blocks: tip_height.0,
+            best_block_hash: GetBestBlockHash(tip_hash),
```

Comment:
Optional: We're using GetBestBlockHash by itself and as part of GetBlockChainInfo. This is a bit confusing, because it's not a get_best_block_hash response here.

Can we rename it based on what it is, rather than how it is used?

```suggestion
            best_block_hash: BlockHashHex(tip_hash),
```